### DISCUSSION TITLE
How to use transmission-daemon or libtransmission?

### DISCUSSION THREAD

**Discussion by ghost** [transmission/transmission#5452] [Q&A]
### What is the issue?

There is literally no documentation.

"one of the most-cited reasons was its easy learning curve"

### Which application of Transmission?

None

**Comment by ckerr** [comment]
https://github.com/transmission/transmission/tree/main/docs is one starting point

**Reply by ghost** [reply]
There is no mention of libtransmission.

Where does transmission-daemon even come from? I have transmission installed on my Mac but I still get `zsh: command not found: transmission-daemon` and nothing is running on port 9091.

**Reply by fredo-47** [reply]
My understanding is that "libtransmission" is just the "library" which comes along with transmission and is used by it.
You do not start "libtransmission" itself, but rather e.g. `transmission-daemon` or one of the other transmission apps.

I don't know about Mac, but when you get `command not found`, it seems you either did not install it properly or it isn't in your PATH which your shell looks for executables in.

**Reply by ghost** [reply]
> My understanding is that "libtransmission" is just the "library" which comes along with transmission and is used by it. 

So libtransmission is an internal thing and I'm not supposed to use it?

>  it seems you either did not install it properly or it isn't in your PATH which your shell looks for executables in.

That's what I'm asking. How do I properly install and use transmission-daemon?

**Reply by ghost** [reply]
I couldn't find any prebuilt transmission-daemon binaries, so I assume I have to build it myself? The instructions from README.md don't work because there is no CMakeLists.txt in the newly created build directory.

**Reply by fredo-47** [reply]
I build in Linux, and the instructions work perfectly well for me.

You wrote 
> there is no CMakeLists.txt in the newly created build directory

But there is a `CMakeLists.txt` in the main directory... 
When you are in the build directory and use the `cmake` command, did you also type the two dots `..` at the end of the command, telling the command that the source is not in the current but the parent directory?

**Reply by ghost** [reply]
> did you also type the two dots `..` at the end of the command, telling the command that the source is not in the current but the parent directory?

Yeah that was the issue, my bad.

**Reply by fredo-47** [reply]
Glad I could help :-)
